rollup/circuitcapacitychecker/impl.go

```diff
@@ -43,8 +47,21 @@ func NewCircuitCapacityChecker(lightMode bool) *CircuitCapacityChecker {
 	return ccc
 }
 
+func NewTempCircuitCapacityChecker() *CircuitCapacityChecker {
+	return &CircuitCapacityChecker{ID: math.MaxUint64}
+}
+
+// IsTemp returns if this CircuitCapacityChecker is backed by a persistent instance
+func (ccc *CircuitCapacityChecker) IsTemp() bool {
+	return ccc.ID == math.MaxUint64
```

Comment:
we should not reuse the same instance

the rust side also maintains some state

Reply:
so you should either create different CircuitCapacityChecker or use some locks

Reply:
>  the rust side also maintains some state

Please see the changes on the rust side as well, math.MaxUint64 is a special ID that forces rust side to not persist any state.

Reply:
ic